test/server/cards/02_SHD/events/ChooseSides.spec.ts

```diff
@@ -0,0 +1,61 @@
+describe('Choose Sides', function() {
+    integration(function(contextRef) {
+        describe('Choose Sides\'s event ability', function() {
+            beforeEach(function() {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['choose-sides'],
+                        spaceArena: ['cartel-spacer'],
+                        leader: { card: 'boba-fett#daimyo', deployed: true },
+                        resources: 30,
+                    },
+                    player2: {
+                        hand: ['cantina-bouncer'],
+                        leader: { card: 'sabine-wren#galvanized-revolutionary', deployed: true },
+                    }
+                });
+            });
+
+            it('should exchange control of a friendly and enemy units', () => {
+                const { context } = contextRef;
+
+                const reset = () => {
+                    context.player1.moveCard(context.chooseSides, 'hand');
+                };
+
+                // Scenario 1: Only a friendly non-leader unit is in play
```

Comment:
This is maybe a case where I'd be supportive of having separate test cases just for the sake of readability, since each case requires different board state